File: CsLox/com/craftinginterpreters/lox/Token.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.craftinginterpreters.lox
{
    /// <summary>
    /// 
    /// </summary>
    internal class Token
    {
        /// <summary>
        /// 
        /// </summary>
        internal TokenType type;

        /// <summary>
        /// 
        /// </summary>
        internal String lexeme;

        /// <summary>
        /// 
        /// </summary>
        internal Object literal;

        /// <summary>
        /// 
        /// </summary>
        internal int line;

        /// <summary>
        /// 
        /// </summary>
        /// <param name="type"></param>
        /// <param name="lexeme"></param>
        /// <param name="literal"></param>
        /// <param name="line"></param>
        internal Token(TokenType type, String lexeme, Object literal, int line)
        {
            this.type = type;
            this.lexeme = lexeme;
            this.literal = literal;
            this.line = line;
        }

        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public override String ToString()
        {
            return type + " " + lexeme + " " + literal;
        }
    }
}

File: JavaLox/src/com/craftinginterpreters/lox/Token.java
package com.craftinginterpreters.lox;

/**
 * A class that represents an identified token in the Lox scripting language.
 * 
 * @author Victor G. Brusca, Carlo Bruscani, based on the work of Robert Nystrom, Copyright Middlemind Games 05/03/2024
 */
class Token {
   /**
    * 
    */
   final TokenType type;
   
   /**
    * 
    */
   final String lexeme;

   /**
    * 
    */
   final Object literal;
   
   /**
    * 
    */
   final int line;

   /**
    * 
    * 
    * @param type
    * @param lexeme
    * @param literal
    * @param line 
    */
   Token(TokenType type, String lexeme, Object literal, int line) {
      this.type = type;
      this.lexeme = lexeme;
      this.literal = literal;
      this.line = line;
   }
   
   /**
    * 
    * @return 
    */
   @Override
   public String toString() {
      return type + " " + lexeme + " " + literal;
   }
}
